public/app/features/canvas/types.ts

```diff
@@ -48,16 +48,20 @@ export interface TextConfig {
   valign: VAlign;
 }
 
-export interface EllipseConfig extends TextConfig {
+export interface CanvasElementConfig extends TextConfig {
   backgroundColor?: ColorDimensionConfig;
+  backgroundImage?: ResourceDimensionConfig;
+  backgroundSize?: BackgroundImageSize;
   borderColor?: ColorDimensionConfig;
-  width?: number;
+  borderWidth?: number;
 }
 
-export interface EllipseData extends TextData {
+export interface CanvasElementData extends TextData {
```

Comment:
We shouldn't need either of these anymore as `rectangle`, `ellipse`, `triangle`, `paralleogram`, and `cloud` should all be based on the `TextConfig` 😬

Reply:
we can disregard this for now, but we probably should figure out maybe a different way of passing the base elements config (i.e. borderWidth, background, etc) down instead of overlaying it with the element specific config / data